### PR TITLE
Email formatting changes

### PR DESCRIPTION
**Pull request by jeabakker**
- unified the site email address fallback to a class function (no more boilerplate) + tests
- the formatting of recipient and sender changed to `Someone <someone@example.com>` instead of just `someone@example.com`

### CODE REVIEW

File: engine/classes/ElggSite.php [hunk 1]
```diff
@@ -335,4 +335,21 @@ public function isPublicPage($url = '') {
...
+	 * Get the email address for the site
+	 *
+	 * This can be set in the basic site settings or fallback to noreply@domain
+	 *
+	 * @return string
+	 * @since 3.0.0
+	 */
+	public function getEmailAddress() {
+		$email = $this->email;
+		if (empty($email)) {
```
Comment (hypeJunction) [inline engine/classes/ElggSite.php]: Should check if it's a valid email perhaps?

File: mod/invitefriends/actions/invitefriends/invite.php [hunk 2]
```diff
@@ -13,6 +13,9 @@
 }
 
 $site = elgg_get_site_entity();
+// create the from address
+$site_email = new \Zend\Mail\Address($site->getEmailAddress(), $site->getDisplayName());
```
Comment (hypeJunction) [inline mod/invitefriends/actions/invitefriends/invite.php]: I think we have an Elgg helper class. We should probably avoid using vendor classes in public API

**Review by mrclay — APPROVED** [review]

**Comment by jdalsem** [comment]
besides the comments this LGTM

File: engine/classes/ElggSite.php (hunk 1 above)
Comment (jeabakker) [inline engine/classes/ElggSite.php]: validity of the email addresses is checked later in the line

**Review by jdalsem — APPROVED** [review]

File: engine/classes/ElggSite.php (hunk 1 above)
Comment (hypeJunction) [inline engine/classes/ElggSite.php]: Where? I would just do:

```php
if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {

}
```

**Review by hypeJunction — APPROVED** [review]

**Pull request merged**